### PR TITLE
FrameDump: Start timing at 0 ticks when starting from boot

### PR DESCRIPTION
**Pull request by JosJuice**
@vadosnaprimer Does this fix your "frame 0" A/V sync problem?

### CODE REVIEW

File: Source/Core/VideoCommon/FrameDump.cpp [hunk 1]
```diff
@@ -447,14 +444,15 @@ void FrameDump::CheckForConfigChange(const FrameData& frame)
   {
     Stop();
     ++m_file_index;
-    PrepareEncoding(frame.width, frame.height);
+    PrepareEncoding(frame.width, frame.height, frame.state.ticks, frame.state.savestate_index);
```
Comment (JosJuice) [inline Source/Core/VideoCommon/FrameDump.cpp:447]: I'm using `frame.state.ticks` here to match master's behavior, but master's behavior seems fishy to me. Doesn't this discard the information about how much time passed between the first frame of this file and the last frame of the previous file? And if so, would the correct thing to use here be the time of the previous frame, or the time of the previous frame plus one "timebase unit"...?

**Comment by vadosnaprimer** [comment]
Can't comment on the code but this patch helped!

File: Source/Core/VideoCommon/FrameDump.cpp [hunk 2]
```diff
@@ -300,7 +297,7 @@ void FrameDump::AddFrame(const FrameData& frame)
     else if (pts > m_context->last_pts + 1 && !m_context->gave_vfr_warning)
     {
       WARN_LOG_FMT(FRAMEDUMP, "PTS delta > 1. Resulting file will have variable frame rate. "
-                              "Subsequent occurances will not be reported.");
+                              "Subsequent occurences will not be reported.");
```
Comment (leoetlino) [inline Source/Core/VideoCommon/FrameDump.cpp:300]:
```suggestion
                              "Subsequent occurrences will not be reported.");
```

File: Source/Core/VideoCommon/FrameDump.cpp (hunk 1 above)
Comment (JosJuice) [reply Source/Core/VideoCommon/FrameDump.cpp:447]: After looking into the ffmpeg API some more, I think the right thing to do would be to explicitly set the duration of the old stream and to use `frame.state.ticks` here. I will leave setting the duration for another PR.

**Review by leoetlino — APPROVED** [review]
The rest of the PR LGTM.

File: Source/Core/VideoCommon/FrameDump.cpp (hunk 1 above)
Comment (JosJuice) [reply Source/Core/VideoCommon/FrameDump.cpp:447]: After looking into this some more, I'm not sure if there's any solution that works great... other than implementing writing audio data to the AVI file, which is a desirable feature regardless of this problem. I think we should move forward with this PR as it is.

**Comment by Pokechu22** [comment]
This PR seems to have [changed FifoCI](https://fifo.ci/version/27712c63218ce4bc225298ce4e97be591147950d/), apparently just due to the introduction of one additional frame.  Is that to be expected?

**Comment by JosJuice** [comment]
No, that's not expected. I'll have to look into this more closely.

**Comment by JosJuice** [comment]
My latest commit fixes the problem with FifoCI reporting duplicate frames, but... I'm not sure if my change makes sense, and the rendering results are different in a way that makes me unsure whether the new behavior or the old behavior is right. If the old behavior is right, I suppose my latest commit is writing to the FIFO too quickly?

**Comment by JosJuice** [comment]
These ones looks a lot better after the change: https://fifo.ci/compare/5726986-5724632/ https://fifo.ci/compare/5727030-5724655/ https://fifo.ci/compare/5727032-5724676/
But this one is just white: https://fifo.ci/compare/5726978-5724709/
This one is missing the shadow: https://fifo.ci/compare/5727093-5724838/
And these ones still have duplicated frames: https://fifo.ci/compare/5727075-5724683/ https://fifo.ci/compare/5727035-5724706/

**Comment by JosJuice** [comment]
I've now removed that commit and placed it in PR #9287 instead.

What I'm thinking is that it would make sense to just set the FifoCI runners to use PNG dumping instead of ffmpeg dumping so that they will completely ignore frame timings. Would this be a reasonable solution? (Of course, we should do it in the config files for the FifoCI runners instead of by changing the default value of `GFX_DUMP_FRAMES_AS_IMAGES` like I've done now for testing.)

**Comment by JosJuice** [comment]
Okay, it seems like FifoCI wasn't set up to handle PNG dumps like I thought...

**Comment by JosJuice** [comment]
@dolphin-emu-bot rebuild

**Comment by Pokechu22** [comment]
@dolphin-emu-bot rebuild

**Comment by JosJuice** [comment]
Now that https://github.com/dolphin-emu/fifoci/pull/34 (and the two PRs before it) is merged, FifoCI is ignoring frame timings and this PR is not causing FifoCI diffs anymore.

**Pull request merged**